### ISSUE TITLE
Controller Functionality Missing

### ISSUE THREAD

**Issue by Death259** [Death259/PiAssist#1]
The entirety of the controller functionality is missing.

At a minimum the script needs to be able to run the RetroArch config. In doing so the script needs to offer the ability to backup the current config. If a backup already exists, then the user needs to be notified.

The retro-arch joyconfig needs to be utilized such that it can configure multiple players (it supports up to 8 players).

**Comment by Death259** [comment]
I've added the initial functionality of the game controller configuration. What currently works, only works with RetroPie.

When the user goes to configure a controller, it asks them if they want to make a backup of the current config. If they say yes and there is currently a backup, it asks if they would like to overwrite the current backup.

**Comment by Death259** [comment]
I've got the basic functionality working, but it looks like the official RetroPie setup script already does this effectively. With that in mind, this will be low on the priority list and not likely to get done anytime soon.

**Comment by MichaelHP** [comment]
I would suggest raising the priority on this.   Main problem I have is controller based.  

I updated to RetoPie V3 Beta 3 and its controller options are extended but buggy and still rely on only single player configuration.      Retroarch itself is having problems setting the configs.      I am using 2x Zero Delay USB encoders and while linux picks them up Retroarch has a hard time finding them correctly.     THe way mapping works is also wonky where it will work for most systems but wont map correctly for FBA.   Since FBA has so many street fighter games where the buttons are mapped is correct.     I would be happy to send you a zero delay encoder board as this is one of the most cost effective solutions yet have had the largest issue getting it to work correctly.

**Comment by MichaelHP** [comment]
It is also incorrectly mapping up  it registers all inputs but up and puts them in the wrong place.

**Comment by Death259** [comment]
I can try to take another look at this, but I believe with beta version 4 the controller configuration issues have been fixed. As for the zero delay encoder board, I'm not even sure I'd know what to do with it, to get it to work. My previous controller configuration is entirely reliant upon the retroarch configurator working, which I believe given the issues it's having means mine won't work either. I've got an idea that I'll try and see if I can make my configuring of controllers not reliant upon retroarch.
